### ISSUE TITLE
IDE0051 false-positive for private copy constructor in records

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn#67910]
**Version Used**: '4.6.0-3.23212.2 (d78a163b)'

**Steps to Reproduce**:

````c#
var a = new A(1);

_ = a with { }; // Use the copy constructor

sealed record A(int Sample)
{
    // IDE0051 private member not used
    private A(A other) => throw new NotImplementedException();
}
````

https://sharplab.io/#v2:CYLg1APgbghgTgAhggvAgdgUwO4IIIAUAjAJQDcAsAFAD6qSC2AlgC4AWCA3ggL6VXUAzphgAbTMARxMAYwD2cSYSboWCAMowAtgAdxJap2oITCAPRmEASQAiAUQAMDgKxEEOuE1gtMCLZi0AI0xEdDk1AFdhYGNTDy8YH3wCPARwthCSVAA+BHY4OVwsXAABIgA6ADlwq11xf1UJOwAPGUwdFiY5dAJyah4gA==

**Diagnostic Id**: IDE0051

**Expected Behavior**: No diagnostic reported

**Actual Behavior**: IDE0051 is reported

**Comment by sdelarosbil** [comment]
@mavasani Hey, can I take this? I took a look at this a bit and I think I have an understanding on how to address this.

**Comment by CyrusNajmabadi** [comment]
Yes. You can take this :-)
